middleware/util/logging/logging-stack-hook.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2018 Shift Devices AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+// NewHook is the initializer for LoggingStackHook{} (implementing logrus.Hook).
+import (
+	"fmt"
+	"strings"
+
+	"github.com/digitalbitbox/bitbox-wallet-app/util/errp"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// stackHook is a logrus hook for logging stack-traces. It implements logrus.Hook.
+type stackHook struct {
+	stackLevels []logrus.Level
+}
+
+// Levels provides the levels to filter.
+func (hook stackHook) Levels() []logrus.Level {
+	return hook.stackLevels
+}
+
+func enrichIfPossible(err error, entry *logrus.Entry) {
```

Comment:
This function seems complex enough (recursively calling itself) that a comment seems warranted; what does `enrich` mean?